pkg/pagerduty/service.go

```diff
@@ -469,15 +472,26 @@ func generatePDServiceDescription(data *Data) string {
 	}
 }
 
-func (c *SvcClient) resolveIncident(serviceKey, incidentKey string) error {
-	event := pdApi.V2Event{}
-	event.Payload = &pdApi.V2Payload{}
-	event.RoutingKey = serviceKey
-	event.Action = "resolve"
-	event.DedupKey = incidentKey
-	event.Payload.Summary = "Cluster does not exist anymore"
-	event.Payload.Source = "pagerduty-operator"
-	event.Payload.Severity = "info"
+// resolveAlert sends an event to the V2 Events API to (eventually) resolve a specific alert.
+// Each service can contain many integration keys, which represent specific integrations
+// enabled for a service. The integration key for the integration that generated the alert
+// identified by the alertKey must be used to successfully delete the alert.
+func (c *SvcClient) resolveAlert(integrationKey, alertKey string) error {
+	event := pdApi.V2Event{
+		RoutingKey: integrationKey,
+		Action:     "resolve",
+		DedupKey:   alertKey,
+		Payload: &pdApi.V2Payload{
+			Summary:  "Cluster does not exist anymore",
+			Source:   "pagerduty-operator",
+			Severity: "info",
+		},
+	}
+
+	// TODO: If the response status is 429 (TooManyRequests) we should probably wait for longer
```

Comment:
Requeue and try again later.  Don't block and wait in the controller.

Reply:
Yes, as soon as all alerts in an incident are resolved, the incident resolves.

Totally agreed on the second point too! I left the TODO for a planned `requeueAfter` some delay if PagerDuty is throttling us.